sbt-version-policy/src/main/scala/sbtversionpolicy/SbtVersionPolicySettings.scala

```diff
@@ -222,27 +222,39 @@ object SbtVersionPolicySettings {
       if (anyError)
         throw new Exception("Compatibility check failed (see messages above)")
     },
-    versionCheck := {
-      val intention =
-        versionPolicyIntention.?.value
-          .getOrElse(throw new MessageOnlyException("Please set the key versionPolicyIntention to declare the compatibility guarantees of this release"))
-      val versionValue = version.value
-      val s            = streams.value
-      val projectId    = thisProject.value.id
-
-      if (Compatibility.isValidVersion(intention, versionValue)) {
-        s.log.info(s"$projectId/$versionValue is a valid version number with respect to the compatibility guarantees '$intention'")
+    versionCheck := Def.taskDyn {
+      if ((publish / skip).value) {
+        Def.task { }
```

Comment:
Dynamic tasks obscure `inspect tree` and worsen parallelism. We should go to great lengths to avoid them.

In this case, for example, we can avoid the `taskDyn` by fetching
```scala
val optIntention = versionPolicyIntention.?.value
```
first, as well as the other values of tasks. And do a regular
```scala
if ((publish / skip).value) {
  ...
}
```
without `Def.taskDyn`.

Reply:
It might be possible to do similar things in the other tasks.

Reply:
I agree, but isn’t this going to evaluate the tasks that we depend on in the two branches of the `if`, regardless of the value of the condition?

Reply:
Yes, but in this case there's nothing but `versionPolicyIntention`, `version`, `streams` and `thisProject`, all of which are safe to evaluate in all cases.

In other places, it might be more difficult. For example the tests might need to be moved to `versionPolicyMimaCheck` and `versionPolicyReportDependencyIssues` rather than in `versionPolicyCheck`.

Reply:
Is it better with `Def.ifS`?

Reply:
I have no idea. I don't exactly what it does. @eed3si9n could you shed some light on this?

Reply:
In sbt 1.4.0 onwards `Def.task { if (x.value) y.value else z.value }`, it would automatically generate conditional task. It does conditional execution but in the inspects it displays both side of the branch - https://eed3si9n.com/selective-functor-in-sbt

Reply:
That is very nice! We should do that, then.